app/facades/places_facade.rb

```diff
@@ -1,9 +1,56 @@
 class PlacesFacade
-  def self.hotels(city_info)
+  def self.places(city_info, categories = nil, page = 0)
+    geoapify_get_places(city_info, categories, page)
+  end
+
+  # PRIVATE METHODS
+  #============================================================================
+
+  def self.search_mapper(places, target_fields)
+    places.map! do |hit|
+      hit.select { |key, _value| target_fields.include?(key) }
+    end
+  end
+
+  #  Geoapify specific methods:
+  #  -------------------------------------------------------------------------
+
+    # Manager method
+  def self.geoapify_get_places(city_info, categories, page)
+    places = geoapify_populate_results(city_info, categories, page)
+    geoapify_filter(places)
+  end
 
-    hotels = PlacesService.get_hotels(city_info)
-    locations = hotels.map do |hotel|
-      Hotel.new(hotel)
+    # Populates results until 20 are present. If a search finds <20, it increases the search radius and tries again.
+    # Accepts pagination, a page value is multiplied by 20 to get the correct result offset, but also adds the number of
+    # past results (if under 20) so results won't be repeated (hopefully).
+  def self.geoapify_populate_results(city_info, categories, page)
+    places = []
+    while places.count < 20
```

Comment:
I would say this is a pretty dangerous approach to solving a search for 20 or more places.

It might be more worthwhile for design to handle this vs programmatically with a potentially infinite loop.

I could image scenarios where weird data does not match anything or even a server issues were results are limited.

This could crash your app and you may have no way to control it if it does.